curator-recipes/src/main/java/org/apache/curator/framework/recipes/cache/TreeCache.java

```diff
@@ -263,11 +281,20 @@ private void doRefreshData() throws Exception
             {
                 if ( dataIsCompressed )
                 {
-                    client.getData().decompressed().usingWatcher(this).inBackground(this).forPath(path);
+                    if (createZkWatches) {
+                        client.getData().decompressed().usingWatcher(this).inBackground(this).forPath(path);
+                    } else {
+                        client.getData().decompressed().inBackground(this).forPath(path);
+                    }
```

Comment:
feel like this could all be a little cleaner with a maybeWatch() helper in the class, e.g.

```
maybeWatch(client.getData().decompressed()).inBackground(this).forPath(path)
```